contracts/NFTStakingManager.sol

```diff
@@ -441,50 +520,94 @@ contract NFTStakingManager is
     _lockTokens(delegationID, tokenIDs);
 
     DelegationInfo storage newDelegation = $.delegations[delegationID];
+    newDelegation.status = DelegatorStatus.PendingAdded;
     newDelegation.owner = owner;
     newDelegation.tokenIDs = tokenIDs;
     newDelegation.validationID = validationID;
+    newDelegation.startingNonce = nonce;
 
     emit InitiatedDelegatorRegistration(validationID, delegationID, tokenIDs);
     return delegationID;
   }
 
   function completeDelegatorRegistration(bytes32 delegationID, uint32 messageIndex) public {
     NFTStakingManagerStorage storage $ = _getNFTStakingManagerStorage();
-    (bytes32 validationID, uint64 nonce) = $.manager.completeValidatorWeightUpdate(messageIndex);
+
     DelegationInfo storage delegation = $.delegations[delegationID];
-    if (validationID != delegation.validationID) {
-      revert validationIDMismatch();
+
+    if (delegation.status != DelegatorStatus.PendingAdded) {
+      revert InvalidDelegatorStatus(delegation.status);
     }
 
+    Validator memory validator = $.manager.getValidator(delegation.validationID);
+
+    if (validator.status == ValidatorStatus.Completed) {
+      revert InvalidValidatorStatus();
+    }
+
+    uint64 nonce;
+
+    if (validator.receivedNonce < delegation.startingNonce) {
+      (bytes32 validationID, uint64 receivedNonce) =
+        $.manager.completeValidatorWeightUpdate(messageIndex);
+      nonce = receivedNonce;
+
+      if (validationID != delegation.validationID) {
+        revert validationIDMismatch();
+      }
+
+      if (nonce < delegation.startingNonce) {
+        revert InvalidNonce(nonce);
+      }
+    }
+
+    delegation.status = DelegatorStatus.Active;
     delegation.startEpoch = getEpochByTimestamp(block.timestamp);
-    emit CompletedDelegatorRegistration(validationID, delegationID, nonce, delegation.startEpoch);
+
+    emit CompletedDelegatorRegistration(
+      delegation.validationID, delegationID, nonce, delegation.startEpoch
+    );
+  }
+
+  function initiateDelegatorRemoval(bytes32[] calldata delegationIDs) external {
```

Comment:
Just have one fn that accepts an array. We are low on contract space and need to conserve.